modules/dnn/src/layers/nary_eltwise_layers.cpp

```diff
@@ -709,8 +709,16 @@ class NaryEltwiseLayerImpl CV_FINAL : public NaryEltwiseLayer
             }
             case OPERATION::MEAN:
             {
-                auto mean = [](const T &a, const T &b) { return (a + b) / T{2}; };
-                nary_forward<T>(mean, T{1} / ninputs, std::forward<Args>(args)...);
+                if (ninputs == 2){
+                    auto mean = [](const T &a, const T &b) { return (a + b) / T{2}; };
+                    binary_forward<T, T>(mean, std::forward<Args>(args)...);
+                }  else if (ninputs == 3) {
+                    auto mean = [](const T &a, const T &b, const T &c) { return (a + b + c) / T{3}; };
+                    trinary_forward<T, T, T, T>(mean, std::forward<Args>(args)...);
```

Comment:
Wrong call.

---

`binary_forward` handles binary operator which takes two operands, such as `Add`, `Sub` ...

`nary_forward` handles operators which can take >= 1 operands, such as `Sum`, `Mean` ...

`trinary_forward` (should be renamed to `ternary_forward` for correct wording) handles operators which take 3 operands, like `Where` ...

They are different types.

Reply:
> @alexlyulkov `nary_forward`does take mean tensor-wise means. Fotensor `x1`, `x2`, `x3` will be calculaled as mean of `x1` and so on. `bi/trinary_forwad` takes batch-wise means for all tensors.

This problem should be fixed like this: https://github.com/opencv/opencv/pull/25630/commits/4be1a1f8d7a80a3d6fed1241801c8f3098223e30